File: migration-kit/contracts/frontend/biRadar.ts
/**
 * BI Radar de Longevidade - contract DTO (espelho do backend
 * FastAPI em migration-kit/backend-integration/schemas/bi_radar.py).
 *
 * Manter este arquivo em sincronia manual com o spec autoritativo
 * em migration-kit/contracts/frontend/biRadar.ts.
 *
 * F1 refactor: clientes do BI sao User de tipo Client em
 * deva_elevveclinic_users. BiRadarClientDto substitui o antigo
 * BiRadarPatientDto.
 *
 * Convencoes:
 * - camelCase para todos os campos (Pydantic emite JSON com o nome
 *   do field por default; field names ja estao em camelCase).
 * - Optional/nullable: description, axisSub, scoreMax, programName
 *   podem ser null.
 * - Envelope de erro segue o contrato canonico (api/errors.py):
 *     { "error": { "status", "code", "message", "details" } }
 */

export type BiRadarPillarDto = {
  pillarCode: string;
  name: string;
  description?: string | null;
  scoreMax?: number | null;
  displayOrder: number;
};

export type BiRadarClientDto = {
  userId: number;
  email: string;
  fullName: string;
  role: string;
  programName?: string | null;
};

export type BiRadarAxisScoreDto = {
  axisKey: string;
  axisLabel: string;
  axisSub?: string | null;
  rawScore: number;
  scoreMax: number;
  clusterActive: boolean;
  clusterBonus: number;
  finalScore: number;
  riskScore: number;
  balanceScore: number;
  classification: string;
  priorityIndex: number;
  isPriorityAxis: boolean;
};

export type BiRadarSummaryDto = {
  risk: number;
  balance: number;
  pendingConfiguration: boolean;
};

export type BiRadarComputationDto = {
  userId: number;
  calculatedAt: string;
  axes: BiRadarAxisScoreDto[];
  priorityAxisKey: string;
  summary: BiRadarSummaryDto;
};

export type BiRadarErrorPayload = {
  error: {
    status: number;
    code: string;
    message: string;
    details: unknown;
  };
};

File: migration-kit/backend-integration/schemas/bi_radar.py
"""Pydantic DTOs para o BI Radar de Longevidade.

Mirror do contract TypeScript em migration-kit/contracts/frontend/biRadar.ts
(espelho manual; nao ha geracao automatica de TS).

Pydantic v2 com field names em camelCase para casar com o JSON
de saida do backend e com o tipo TypeScript do frontend.

F1 refactor: clientes do BI sao User de tipo Client. BiRadarClientDto
substitui o antigo BiRadarPatientDto.
"""
from __future__ import annotations

from typing import Optional

from pydantic import BaseModel


class BiRadarPillarDto(BaseModel):
    """Metadado de um pilar (usado em /bi/radar/pillars)."""
    pillarCode: str
    name: str
    description: Optional[str] = None
    scoreMax: Optional[int] = None
    displayOrder: int


class BiRadarClientDto(BaseModel):
    """Cliente (User de tipo Client) usado no BI Radar.

    F1 refactor: substitui BiRadarPatientDto. Os campos sao derivados
    de deva_elevveclinic_users (role='client').
    """
    userId: int
    email: str
    fullName: str
    role: str  # sempre "client" para este DTO
    programName: Optional[str] = None


class BiRadarAxisScoreDto(BaseModel):
    """Score calculado de 1 pilar."""
    axisKey: str
    axisLabel: str
    axisSub: Optional[str] = None
    rawScore: float
    scoreMax: float
    clusterActive: bool
    clusterBonus: float
    finalScore: float
    riskScore: float
    balanceScore: float
    classification: str
    priorityIndex: float
    isPriorityAxis: bool


class BiRadarSummaryDto(BaseModel):
    """Resumo agregado do calculo."""
    risk: float
    balance: float
    pendingConfiguration: bool


class BiRadarComputationDto(BaseModel):
    """Resposta completa do calculo do Radar para 1 cliente."""
    userId: int
    calculatedAt: str
    axes: list[BiRadarAxisScoreDto]
    priorityAxisKey: str
    summary: BiRadarSummaryDto
